XamarinCommunityToolkit/Behaviors/UserStoppedTypingBehavior.shared.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+using System.Windows.Input;
+using Xamarin.Forms;
+
+namespace Xamarin.CommunityToolkit.Behaviors
+{
+	public class UserStoppedTypingBehavior : BaseBehavior
+	{
+		CancellationTokenSource tokenSource;
+
+		#region Bindable Properties
+
+		public static readonly BindableProperty CommandProperty = BindableProperty.Create(nameof(Command), typeof(ICommand), typeof(UserStoppedTypingBehavior));
+		public ICommand Command
+		{
+			get => (ICommand)GetValue(CommandProperty);
+			set => SetValue(CommandProperty, value);
+		}
+
+		public static readonly BindableProperty StoppedTypingTimeThresholdProperty = BindableProperty.Create(nameof(StoppedTypingTimeThreshold), typeof(int), typeof(UserStoppedTypingBehavior), 1000);
+		public int StoppedTypingTimeThreshold
+		{
+			get => (int)GetValue(StoppedTypingTimeThresholdProperty);
+			set => SetValue(StoppedTypingTimeThresholdProperty, value);
+		}
+
+		public static readonly BindableProperty ShouldDismissKeyboardAutomaticallyProperty = BindableProperty.Create(nameof(ShouldDismissKeyboardAutomatically), typeof(bool), typeof(UserStoppedTypingBehavior), false);
+		public bool ShouldDismissKeyboardAutomatically
+		{
+			get => (bool)GetValue(ShouldDismissKeyboardAutomaticallyProperty);
+			set => SetValue(ShouldDismissKeyboardAutomaticallyProperty, value);
+		}
+
+		#endregion Bindable Properties
+
+		protected override void OnAttachedTo(View view)
+		{
+			base.OnAttachedTo(view);
+
+			if (!(view is InputView inputView)) return;
+
+			inputView.TextChanged += InputView_TextChanged;
+		}
+
+		protected override void OnDetachingFrom(View view)
+		{
+			base.OnDetachingFrom(view);
+
+			if (!(view is InputView inputView)) return;
+
+			inputView.TextChanged -= InputView_TextChanged;
+		}
+
+		void InputView_TextChanged(object sender, TextChangedEventArgs e)
+		{
+			if (tokenSource != null)
+				tokenSource.Cancel();
+
+			tokenSource = new CancellationTokenSource();
+
+#pragma warning disable CS4014 // Because this call is not awaited, execution of the current method continues before the call is completed
+			PerformTextChanged(sender as Entry, e.NewTextValue, tokenSource.Token);
```

Comment:
Also, instead of using `sender as Entry` can you use `(Entry)sender`. Because if the View isn't an `Entry` we can face a null exception at `entry.Unfocu() //line 74`. So I think it's better to fail here because of the sync/async behavior. Do you agree?

Reply:
Hi @pictos, actually the type check has already been done in the OnAttached() method. So that is not possible. But I did see that I was casting to an Entry instead of an InputView, so will change that here.